pkg/plugin/validator_test.go

```diff
@@ -0,0 +1,211 @@
+				GitHubAppID:             "test-github-id",
+				GitHubAppInstallationID: "test-install-id",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			fakeTokenServerResqCode: http.StatusCreated,
+			wantErrSubstr:           "invalid issue url",
+			issueBytes:              []byte(`{"state": "open"}`),
+		},
+		{
+			name:     "issue_not_int",
+			issueURL: "https://github.com/test-owner/test-repo/issues/abc",
+			cfg: &PluginConfig{
+				GitHubAppID:             "test-github-id",
+				GitHubAppInstallationID: "test-install-id",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			fakeTokenServerResqCode: http.StatusCreated,
+			wantErrSubstr:           "invalid issue url, issueURL doesn't match pattern",
+			issueBytes:              []byte(`{"state": "open"}`),
+		},
+		{
+			name:     "unauthorized",
+			issueURL: "https://github.com/test-owner/test-repo/issues/1",
+			cfg: &PluginConfig{
+				GitHubAppID:             "test-github-id",
+				GitHubAppInstallationID: "test-install-id",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			fakeTokenServerResqCode: http.StatusUnauthorized,
+			wantErrSubstr:           "failed to get access token",
+			issueBytes:              []byte(`{"state": "open"}`),
+		},
+		{
+			name:     "issue_not_open",
+			issueURL: "https://github.com/test-owner/test-repo/issues/1",
+			cfg: &PluginConfig{
+				GitHubAppID:             "test-github-id",
+				GitHubAppInstallationID: "test-install-id",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			fakeTokenServerResqCode: http.StatusCreated,
+			wantErrSubstr:           "issue is in state: closed",
+			issueBytes:              []byte(`{"state": "closed"}`),
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		ctx := context.Background()
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			fakeTokenServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if r.Header.Get("Accept") != "application/vnd.github+json" {
+					w.WriteHeader(500)
+					fmt.Fprintf(w, "missing accept header")
+					return
+				}
+				authHeader := r.Header.Get("Authorization")
+				if authHeader == "" || !strings.HasPrefix(authHeader, "Bearer ") {
+					w.WriteHeader(http.StatusInternalServerError)
+					fmt.Fprintf(w, "missing or malformed authorization header")
+					return
+				}
+				w.WriteHeader(tc.fakeTokenServerResqCode)
+				fmt.Fprintf(w, `{"token":"this-is-the-token-from-github"}`)
+			}))
+			defer fakeTokenServer.Close()
+
+			hc := newTestServer(t, testHandleIssueReturn(t, tc.issueBytes))
+			testGitHubClient := github.NewClient(hc)
+
+			ghAppOpts := []githubapp.ConfigOption{
+				githubapp.WithJWTTokenCaching(1 * time.Minute),
+				githubapp.WithAccessTokenURLPattern(fakeTokenServer.URL + "/%s/access_tokens"),
+			}
+			testGHAppCfg := githubapp.NewConfig(tc.cfg.GitHubAppID, tc.cfg.GitHubAppInstallationID, testPrivateKey, ghAppOpts...)
+			testGituhbApp := githubapp.New(testGHAppCfg)
```

Comment:
nit: close githubapp client?

Reply:
http client doesn't need to be closed. https://stackoverflow.com/questions/36688633/how-to-release-http-client-in-go